aci/resource_aci_fabricnodeidentp.go

```diff
@@ -263,35 +312,117 @@ func resourceAciFabricNodeMemberRead(ctx context.Context, d *schema.ResourceData
 	log.Printf("[DEBUG] %s: Beginning Read", d.Id())
 
 	aciClient := m.(*client.Client)
-
 	dn := d.Id()
 	fabricNodeIdentP, err := getRemoteFabricNodeMember(aciClient, dn)
-
 	if err != nil {
 		return errorForObjectNotFound(err, dn, d)
 	}
+
+	getAndsetDecommissionedNodes(aciClient, d.Get("serial").(string), d)
+
 	_, err = setFabricNodeMemberAttributes(fabricNodeIdentP, d)
 	if err != nil {
 		d.SetId("")
 		return nil
 	}
+
 	log.Printf("[DEBUG] %s: Read finished successfully", d.Id())
 
 	return nil
 }
 
 func resourceAciFabricNodeMemberDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	log.Printf("[DEBUG] %s: Beginning Destroy", d.Id())
-
 	aciClient := m.(*client.Client)
 	dn := d.Id()
-	err := aciClient.DeleteByDn(dn, "fabricNodeIdentP")
+
+	attachedToSwitch, err := verifyNodeAttachedToSwitch(aciClient, d.Get("serial").(string))
 	if err != nil {
 		return diag.FromErr(err)
 	}
+	if attachedToSwitch {
+		decommissionFabricNodeMemberFromController("true", "no", d, m)
+	} else {
+		err = aciClient.DeleteByDn(dn, "fabricNodeIdentP")
+		if err != nil {
+			return diag.FromErr(err)
```

Comment:
just for my understanding why is this not always called at destroy operation?

Reply:
I a node is attcahed to the fabric, then we call 

which in turn does
"fabricRsDecommissionNode": {
			"attributes": {
				"tDn": "topology/pod-%s/node-%s",
				"status": "created,modified",
				"removeFromController": "true" -> when it's true it deletes the node after decommission, i don't need to delete again manually.